enigma-principal/app/src/epoch_u/epoch_provider.rs

```diff
@@ -56,146 +54,255 @@ impl EpochProvider {
     }
 
     #[logfn(DEBUG)]
-    pub fn read_epoch_state() -> Result<Option<EpochState>, Error> {
-        let epoch_state = match File::open(Self::get_state_file_path()?) {
+    fn read_from_file(cap: usize) -> Result<Vec<EpochState>, Error> {
+        let epoch_state = match File::open(Self::get_file_path()?) {
             Ok(mut f) => {
                 let mut buf = Vec::new();
                 f.read_to_end(&mut buf)?;
                 let mut des = Deserializer::new(&buf[..]);
-                let epoch_state: Option<EpochState> = match Deserialize::deserialize(&mut des) {
-                    Ok(value) => Some(value),
+                let mut data: Vec<EpochState> = match Deserialize::deserialize(&mut des) {
```

Comment:
btw, here you can use `unwrap_or_default()` which will make it way more readable.

Reply:
Fixed 24a61b09ab7ab77aa12f6722aa7f23cae3ad514d